ground/src/main/java/com/google/android/ground/util/Debug.kt

```diff
@@ -15,13 +15,19 @@
  */
 package com.google.android.ground.util
 
+import com.google.android.ground.FirebaseCrashLogging
 import timber.log.Timber
 
 object Debug {
   fun logLifecycleEvent(instance: Any) {
     val stackTrace = Thread.currentThread().stackTrace
     val callingMethod = stackTrace[3].methodName + "()"
-    Timber.tag(instance.javaClass.simpleName).v("Lifecycle event: $callingMethod")
+    val className = instance.javaClass.simpleName
```

Comment:
This method relies on us remembering to call `logLifecycleEvent()` directly or via `super()`. Instead, can we use [registerActivityLifecycleCallbacks](https://developer.android.com/reference/android/app/Application.html#registerActivityLifecycleCallbacks) to do the same? We should probably move lifecycle event logging there as well.

Reply:
Thanks Gino! It looks like this callback only gets notified for activity lifecycle changes, and not fragment.